src/Mvc/Extensions.ApiDescription.Client/src/build/Microsoft.Extensions.ApiDescription.Client.targets

```diff
@@ -30,19 +27,16 @@
     </ProjectReference>
   </ItemGroup>
 
-  <Target Name="_CreateOpenApiReferenceItemsForOpenApiProjectReferences"
-      AfterTargets="ResolveProjectReferences"
-      Condition="'$(DesignTimeBuild)' != 'true' AND '$(BuildingProject)' == 'true'">
+  <Target Name="_CreateOpenApiReferenceItemsForOpenApiProjectReferences" DependsOnTargets="ResolveProjectReferences">
     <ItemGroup>
       <_Temporary Remove="@(_Temporary)" />
     </ItemGroup>
 
     <MSBuild Targets="OpenApiGetDocuments"
-        Projects="@(ProjectReferenceWithConfiguration)"
-        Condition="'%(ProjectReferenceWithConfiguration.OpenApiReference)' == 'true'"
-        Properties="%(ProjectReferenceWithConfiguration.SetConfiguration); %(ProjectReferenceWithConfiguration.SetPlatform); %(ProjectReferenceWithConfiguration.SetTargetFramework)"
+        BuildInParallel="$(BuildInParallel)"
+        Projects="@(OpenApiProjectReference)"
         RebaseOutputs="true"
-        RemoveProperties="%(ProjectReferenceWithConfiguration.GlobalPropertiesToRemove);TargetFrameworks;RuntimeIdentifier">
+        RemoveProperties="Configuration;Platform;RuntimeIdentifier;TargetFramework;TargetFrameworks">
```

Comment:
@rainersigwald is removing these properties sufficient?

And, am I missing something important in the shift from `@(ProjectReferenceWithConfiguration)` to using `@(OpenApiProjectReference)` items directly? (Without this, design-time builds would still have been broken and `'$(OpenApiBuildReferencedProjects)' == 'false'` would not work.)

Reply:
I'd definitely want to keep something like `%(OpenApiProjectReference.GlobalPropertiesToRemove)` in there as an escape hatch. This list looks ok to me, but there's always a risk of things changing (like when multitargeting was added) or user projects having their own custom dimensions.

Reply:
Great suggestion! I added this to #4923 because I have other bits to do short-term.